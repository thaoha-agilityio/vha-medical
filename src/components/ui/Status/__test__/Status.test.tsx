import { render } from '@testing-library/react';
import { Status, StatusProps } from '..';

describe('Status test cases', () => {
  const mockProps: StatusProps = {
    status: 0,
    className: '',
  };
  const setup = (props: StatusProps) => render(<Status {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const { asFragment } = setup(mockProps);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const { asFragment } = setup({ status: 1, className: 'mock' });

    expect(asFragment()).toMatchSnapshot();
  });
});
