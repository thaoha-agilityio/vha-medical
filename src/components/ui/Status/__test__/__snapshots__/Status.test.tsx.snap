// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status test cases should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-success text-white dark:text-success dark:bg-opacity-20 w-[68px] px-2 rounded-md text-center"
  >
    <p
      class="text-2xs font-semibold capitalize"
    >
      new
    </p>
  </div>
</DocumentFragment>
`;

exports[`Status test cases should render with provided props 1`] = `
<DocumentFragment>
  <div
    class="bg-warning text-white dark:text-warning dark:bg-opacity-20 w-[68px] px-2 rounded-md text-center mock"
  >
    <p
      class="text-2xs font-semibold capitalize"
    >
      meeting
    </p>
  </div>
</DocumentFragment>
`;
